### PR TITLE
Result component and a whole lot more...

### PR DESCRIPTION
**Pull request by JohanAhlen**
- Editor and notebook split
- NPM module improvements
- Result grid v1
- Externalized sql reference from the editor

### CODE REVIEW

File: desktop/core/src/desktop/js/apps/editor/EditorViewModel.js [hunk 1]
```diff
@@ -23,11 +23,11 @@ import ChartTransformers from 'apps/notebook/chartTransformers';
 import huePubSub from 'utils/huePubSub';
 import hueUtils from 'utils/hueUtils';
 
-import Notebook from 'apps/notebook2/notebook';
+import Notebook from 'apps/editor/notebook';
```
Comment (romainr) [inline desktop/core/src/desktop/js/apps/editor/EditorViewModel.js:26]: Nice!

File: desktop/core/src/desktop/js/components/HueTable.test.ts [hunk 2]
```diff
@@ -39,6 +39,26 @@ describe('HueTable.vue', () => {
     expect(wrapper.element).toMatchSnapshot();
   });
 
+  it('should render a table with array rows', () => {
```
Comment (romainr) [inline desktop/core/src/desktop/js/components/HueTable.test.ts:42]: Nice!

File: desktop/core/src/desktop/js/components/HueTable.test.ts [hunk 3]
```diff
@@ -79,4 +99,29 @@ describe('HueTable.vue', () => {
     });
     expect(wrapper.element).toMatchSnapshot();
   });
+
+  it('should render a stickt table with header and column css', () => {
```
Comment (romainr) [inline desktop/core/src/desktop/js/components/HueTable.test.ts:103]: stickt --> sticky or SMD

File: desktop/core/src/desktop/js/components/__snapshots__/HueTable.test.ts.snap [hunk 4]
```diff
@@ -1,5 +1,132 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
+exports[`HueTable.vue should render a stickt table with header and column css 1`] = `
```
Comment (romainr) [inline desktop/core/src/desktop/js/components/__snapshots__/HueTable.test.ts.snap:3]: Or stickt real name?

File: desktop/libs/notebook/src/notebook/templates/notebook.mako [hunk 5]
```diff
@@ -19,55 +19,15 @@
...
 
 <%namespace name="configKoComponents" file="/config_ko_components.mako" />
-<%namespace name="editorComponents2" file="editor_components2.mako" />
 <%namespace name="editorComponents" file="editor_components.mako" />
 <%namespace name="notebookKoComponents" file="/common_notebook_ko_components.mako" />
 <%namespace name="hueAceAutocompleter" file="/hue_ace_autocompleter.mako" />
 
 
 <div id="notebookComponents" class="notebook">
-% if ENABLE_NOTEBOOK_2.get():
```
Comment (romainr) [inline desktop/libs/notebook/src/notebook/templates/notebook.mako:33]: Nice

File: docs/docs-site/content/developer/components/_index.md [hunk 6]
```diff
@@ -31,7 +31,7 @@ To run the [demo app](https://github.com/cloudera/hue/tree/master/tools/examples
 2. In [package.json](https://github.com/cloudera/hue/blob/master/tools/examples/api/hue_dep/package.json), remove `"hue": "file:../../../.."` without touching the newly added gethue dependency
 3. In [src/app.js](https://github.com/cloudera/hue/blob/master/tools/examples/api/hue_dep/src/app.js), change the import line to:
 
-       import sqlAutocompleteParser from 'gethue/parse/sql/hive/hiveAutocompleteParser';
+       import sqlAutocompleteParser from 'gethue/lib/parsers/hiveAutocompleteParser';
```
Comment (romainr) [inline docs/docs-site/content/developer/components/_index.md:34]: Bueno!

Might break current ER in 'Relationships' tab in Table browser, to check after

File: docs/docs-site/content/developer/components/_index.md [hunk 7]
```diff
@@ -84,4 +84,4 @@ Please refer these [demo apps](https://github.com/cloudera/hue/tree/master/tools
 
 Internally these components are created using Vue.js & TypeScript. So you can even use them as plain Vue component, and take advantage of Vue features. Vue version of the components are under `gethue/components`.
 
-    import ERDiagram from 'gethue/components/er-diagram/index.vue';
+    import ERDiagram from 'gethue/lib/components/er-diagram/index.vue';
```
Comment (romainr) [inline docs/docs-site/content/developer/components/_index.md:87]: And we can republish the lib after 4.9

**Review by romainr — APPROVED** [review]
Nice!

And CI should be verde after https://github.com/cloudera/hue/pull/1692

File: desktop/core/src/desktop/js/components/HueTable.test.ts (hunk 3 above)
Comment (JohanAhlen) [reply desktop/core/src/desktop/js/components/HueTable.test.ts:103]: should be SMD indeed.

File: docs/docs-site/content/developer/components/_index.md (hunk 7 above)
Comment (JohanAhlen) [reply docs/docs-site/content/developer/components/_index.md:87]: Ah, this shouldn't be "lib", lib is for pre-packaged components, but here it's using the vue version. Will fix.

**Pull request closed without merging**